scalanet/src/io/iohk/scalanet/codec/FramingCodec.scala

```diff
@@ -0,0 +1,79 @@
+package io.iohk.scalanet.codec
+
+import java.nio.ByteBuffer
+
+import io.iohk.decco.{BufferInstantiator, Codec}
+import io.iohk.decco.Codec.{DecodeResult, Failure}
+import io.iohk.scalanet.codec.FramingCodec.State._
+import io.iohk.scalanet.codec.FramingCodec._
+
+import scala.collection.mutable
+
+class FramingCodec[T](messageCodec: Codec[T]) extends StreamCodec[T] {
+
+  private var state: State = LengthExpected
+  private var length: Int = 0
+  private var db: ByteBuffer = _
+  private val nlb = ByteBuffer.allocate(4)
+
+  override def streamDecode[B](source: B)(implicit bi: BufferInstantiator[B]): Seq[T] = this.synchronized {
+    val b = bi.asByteBuffer(source)
+    val s = mutable.ListBuffer[T]()
+    while (b.remaining() > 0) {
+      state match {
+        case LengthExpected =>
```

Comment:
Wasn't this what the `LengthFieldPrepender` was doing and the task was supposed to avoid?

Reply:
The idea is that the delimiting is not hard-coded in TCPPeerGroup so that now the user has complete control over the wire format.